TESTS/host_tests/pyusb_basic.py

```diff
@@ -0,0 +1,333 @@
+            if  ep_type == 2:
+                if endpoint.bEndpointAddress & 0x80:
+                    assert bulk_in is None
+                    bulk_in = endpoint
+                else:
+                    assert bulk_out is None
+                    bulk_out = endpoint
+            elif ep_type == 3:
+                if endpoint.bEndpointAddress & 0x80:
+                    assert int_in is None
+                    int_in = endpoint
+                else:
+                    assert int_out is None
+                    int_out = endpoint
+        assert bulk_in is not None
+        assert bulk_out is not None
+        assert int_in is not None
+        assert int_out is not None
+        bulk_out.write("hello" + "x" *256);
+        int_out.write("world" + "x" *256);
+
+        dev.set_interface_altsetting(0, 1)
+
+        intf = get_interface(dev, 0, 0)
+
+        # Find endpoints
+        bulk_in = None
+        bulk_out = None
+        int_in = None
+        int_out = None
+
+        for endpoint in intf:
+            log("Processing endpoint %s" % endpoint)
+            ep_type = endpoint.bmAttributes & 0x3
+            if  ep_type == 2:
+                if endpoint.bEndpointAddress & 0x80:
+                    assert bulk_in is None
+                    bulk_in = endpoint
+                else:
+                    assert bulk_out is None
+                    bulk_out = endpoint
+            elif ep_type == 3:
+                if endpoint.bEndpointAddress & 0x80:
+                    assert int_in is None
+                    int_in = endpoint
+                else:
+                    assert int_out is None
+                    int_out = endpoint
+        assert bulk_in is not None
+        assert bulk_out is not None
+        assert int_in is not None
+        assert int_out is not None
+        bulk_out.write("hello2" + "x" *256);
+        int_out.write("world2" + "x" *256);
+
+
+        t = Thread(target=write_data, args=(bulk_out,))
+        t.start()
+
+        for _ in range(10):
+            request_type = 0x40                         # Host-to-device, Vendor request to Device
+            request = VENDOR_TEST_CTRL_NONE_DELAY       # Vendor custom request
+            value = 0                                   # Always 0 for this request
+            index = 0                                   # Communication interface
+            length = 0                                  # No data
+            dev.ctrl_transfer(request_type, request, value, index, length, 5000)
+
+        t.join()
+
+        return True
+
+def write_data(pipe):
+    print("Write data running")
+    count = 0
+    for _ in range(40):
+        pipe.write("Value is %s" % count)
+        count += 1
+        print("Count %s" % count)
+        time.sleep(0.5)
+    
+    
+def control_stall_test(dev, log):
+
+    # Control OUT stall
+    try:
+        request_type = 0x40                         # Host-to-device, Vendor request to Device
```

Comment:
would be better to have less magic numbers, we can use `usb.util` stuff
```
# Host-to-device, Vendor request to Device
request_type = usb.util.build_request_type(
                                usb.util.CTRL_OUT, 
                                usb.util.CTRL_TYPE_VENDOR, 
                                usb.util.CTRL_RECIPIENT_DEVICE)
```

Reply:
I would say the less magic numbers there are, the better.

Reply:
done